dotnet/src/Extensions/Experimental/Orchestration.Flow/Model/FlowStep.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.SemanticKernel.Experimental.Orchestration.Execution;
+
+#pragma warning disable IDE0130
+namespace Microsoft.SemanticKernel.Experimental.Orchestration;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// Step within a <see cref="Flow"/> which defines the step goal, available plugins, required and provided variables.
+/// </summary>
+public class FlowStep
```

Comment:
Should this be sealed?

Reply:
no, for example `Flow` can also be a `FlowStep`